examples/cd_text_amazon.ipynb

```diff
@@ -0,0 +1,553 @@
+    "ds_h0 = Subset(ds_tr, idx_h0)\n",
+    "ds_h1 = ds.get_subset('test')\n",
+    "dl = partial(DataLoader, shuffle=True, batch_size=100, collate_fn=ds.collate, num_workers=2)\n",
+    "dl_ref, dl_h0, dl_h1 = dl(ds_ref), dl(ds_h0), dl(ds_h1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Detect drift\n",
+    "\n",
+    "### MMD detector on transformer embeddings\n",
+    "\n",
+    "First we embed instances using a pretrained transformer model and detect data drift using the [MMD detector](https://docs.seldon.io/projects/alibi-detect/en/latest/methods/mmddrift.html) on the embeddings.\n",
+    "\n",
+    "Helper functions:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import List\n",
+    "\n",
+    "\n",
+    "def update_flat_list(x: List[list]):\n",
+    "    return [item for sublist in x for item in sublist]\n",
+    "\n",
+    "\n",
+    "def accumulate_sample(dataloader: DataLoader, sample_size: int, stars: int = None):\n",
+    "    \"\"\" Create batches of data from dataloaders. \"\"\"\n",
+    "    batch_count, stars_count = 0, 0\n",
+    "    x_out, y_out, meta_out = [], [], []\n",
+    "    for x, y, meta in dataloader:\n",
+    "        y, meta = y.numpy(), meta.numpy()\n",
+    "        if isinstance(stars, int):\n",
+    "            idx_stars = np.where(y == stars)[0]\n",
+    "            y, meta = y[idx_stars], meta[idx_stars]\n",
+    "            x = tuple([x[idx] for idx in idx_stars])\n",
+    "        n_batch = y.shape[0]\n",
+    "        idx = min(sample_size - batch_count, n_batch)\n",
+    "        batch_count += n_batch\n",
+    "        x_out += [x[:idx]]\n",
+    "        y_out += [y[:idx]]\n",
+    "        meta_out += [meta[:idx]]\n",
+    "        if batch_count >= sample_size:\n",
+    "            break\n",
+    "    x_out = update_flat_list(x_out)\n",
+    "    y_out = np.concatenate(y_out, axis=0)\n",
+    "    meta_out = np.concatenate(meta_out, axis=0)\n",
+    "    return x_out, y_out, meta_out"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Define the transformer embedding preprocessing step:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from alibi_detect.cd import MMDDrift\n",
+    "from alibi_detect.cd.pytorch import preprocess_drift\n",
+    "from alibi_detect.models.pytorch import TransformerEmbedding\n",
+    "from functools import partial\n",
+    "from transformers import AutoTokenizer\n",
+    "\n",
+    "emb_type = 'hidden_state'  # pooler_output, last_hidden_state or hidden_state\n",
+    "# layers to extract hidden states from for the embedding used in drift detection\n",
+    "# only relevant for emb_type = 'hidden_state'\n",
+    "n_layers = 8\n",
+    "layers = [-_ for _ in range(1, n_layers + 1)]\n",
```

Comment:
Is the `-` sign here intentional?

Reply:
Yes, using the last `n_layers` hidden layers.